library/spdm_secured_message_lib/libspdm_secmes_session.c

```diff
@@ -595,21 +616,27 @@ spdm_create_update_session_data_key(IN void *spdm_secured_message_context,
                 .request_data_secret,
             hash_size);
         ASSERT(ret_val);
+        if (!ret_val) {
+            return RETURN_DEVICE_ERROR;
+        }
         DEBUG((DEBUG_INFO, "RequestDataSecretUpdate (0x%x) - ",
                hash_size));
         internal_dump_data(secured_message_context->application_secret
                        .request_data_secret,
                    hash_size);
         DEBUG((DEBUG_INFO, "\n"));
 
-        spdm_generate_aead_key_and_iv(
+        status = spdm_generate_aead_key_and_iv(
             secured_message_context,
             secured_message_context->application_secret
                 .request_data_secret,
             secured_message_context->application_secret
                 .request_data_encryption_key,
             secured_message_context->application_secret
                 .request_data_salt);
+        if (RETURN_ERROR(status)) {
+            return status;
+        }
         secured_message_context->application_secret
             .request_data_sequence_number = 0;
```

Comment:
@jyao1 or others - there's the possibility here that we want to update both keys, and updating requester keys succeeds but updating responder keys fails. It would leave us in a tricky situation to determine whether we want to say the requester backup is valid.

It looks like we would only ever call this function to update the requester or update the responder - we're only doing one at a time right now. But if we want to avoid having a state issue, we should probably make this an 

`   else if ((action & SPDM_KEY_UPDATE_ACTION_RESPONDER) != 0) {`

so that we don't have to worry about a halfway complete operation.

Reply:
halfway completion is headache.
Since we only call this function to update one party, how about we make it mandatory?
We can assert only SPDM_KEY_UPDATE_ACTION_RESPONDER or SPDM_KEY_UPDATE_ACTION_REQUESTER is called. Then we dont need worry this case.

I try to *avoid* the complicated situation, instead of *resolve* it.

Reply:
Agreed, which is why I figured we change the function slightly such that it is exclusive to one or other actions, but not both. I can do in this PR.

Reply:
Updated to resolve. We probably deserve a separate item/investigation for how we handle halfway completed results elsewhere as well.